### PR TITLE
Remove ngraph::Lambda class, replace TensorIterator body with ngraph::Function

### PR DESCRIPTION
**Pull request by itikhono**
Removed `ngraph::Lambda` class.
Moved the functionality to `ngraph::Function`
Replaced use of `ngraph::Lambda` with `ngraph::Function`

After removing Lambda it turned out that passing parameters (not initializes) from parent graph to subgraph does not work correctly (as a result a few unit tests fail).  Necessary changes were made by @mbencer in this PR https://github.com/itikhono/openvino/pull/1

### CODE REVIEW

File: ngraph/core/include/ngraph/lambda.hpp [hunk 1]
```diff
@@ -1,69 +0,0 @@
...
-#include "ngraph/node.hpp"
-#include "ngraph/op/parameter.hpp"
-#include "ngraph/op/result.hpp"
-
-namespace ngraph
-{
-    class NGRAPH_API Lambda
-    {
-    public:
-        virtual ~Lambda() {}
```
Comment (GlebKazantaev) [inline ngraph/core/include/ngraph/lambda.hpp:29]: Please keep virtual dtor for Function

File: ngraph/core/include/ngraph/function.hpp [hunk 2]
```diff
@@ -120,6 +120,24 @@ namespace ngraph
...
         void set_topological_sort(topological_sort_t);
 
+        virtual bool visit_attributes(AttributeVisitor& visitor);
+
+        /// Return the function parameters
+        const ParameterVector& get_parameters() const { return m_parameters; };
+        /// Return a list of function's outputs
+        const ResultVector& get_results() const { return m_results; };
+        /// Index for parameter, or -1
+        int64_t get_parameter_index(const std::shared_ptr<op::Parameter>& parameter) const;
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/function.hpp:130]: Do we use this method?

File: ngraph/core/include/ngraph/function.hpp [hunk 3]
```diff
@@ -120,6 +120,24 @@ namespace ngraph
...
+
+        /// Index for value or result referencing it, or -1
+        int64_t get_result_index(const Output<Node>& value) const;
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/function.hpp:133]: And this?

File: ngraph/core/include/ngraph/function.hpp (hunk 2 above)
Comment (GlebKazantaev) [reply ngraph/core/include/ngraph/function.hpp:130]: They are needed for searching Parameters in Function

File: ngraph/core/include/ngraph/node.hpp [hunk 4]
```diff
@@ -100,6 +100,7 @@ namespace ngraph
     NGRAPH_API
     NodeVector as_node_vector(const OutputVector& values);
     /// Returns a ResultVector referencing values.
+    NGRAPH_API
```
Comment (mbencer) [inline ngraph/core/include/ngraph/node.hpp:103]: @ilyachur Is it ok to expose `as_result_vector`? (it is used in onnx importer)

Comment (ilyachur) [reply ngraph/core/include/ngraph/node.hpp:103]: I am ok this this change, if it is required for external components.

**Review by tomdol — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

File: ngraph/core/include/ngraph/lambda.hpp (hunk 1 above)
Comment (itikhono) [reply ngraph/core/include/ngraph/lambda.hpp:29]: unnecessary, no inheritance from function

**Pull request merged**